src/components/switchComponent/index.tsx

```diff
@@ -1,17 +1,49 @@
-import React, { FC } from 'react';
+import React, { FC, ReactNode } from 'react';
+import { FlexboxProps } from '@chakra-ui/styled-system';
 import {
   Switch as ChakraSwitch,
   SwitchProps as ChakraSwitchProps,
+  FormControl,
+  FormLabel,
 } from '@chakra-ui/react';
 
 export type SwitchProps = Pick<
   ChakraSwitchProps,
-  'onChange' | 'isChecked' | 'variant'
->;
+  'onChange' | 'isChecked' | 'variant' | 'isDisabled'
+> & {
+  label?: ReactNode;
+  labelPosition?: FlexboxProps['flexDirection'];
+};
 
-const Switch: FC<SwitchProps> = ({ onChange, isChecked, variant }) => {
+const Switch: FC<SwitchProps> = ({
+  onChange,
+  isChecked,
+  variant,
+  isDisabled,
+  label,
+  labelPosition = 'row',
+}) => {
   return (
-    <ChakraSwitch onChange={onChange} isChecked={isChecked} variant={variant} />
+    <FormControl
+      width="min-content"
+      display="flex"
+      alignItems="center"
+      gap="sm"
```

Comment:
flex-gap has poor browser support and we should not use it. Can you use margin or grid instead?